### DISCUSSION TITLE
Recommended upgrade frequency?

### DISCUSSION THREAD

**Discussion by ghost** [Homebrew/discussions#5112] [Everyday usage] [answered]
### Output of `brew config`

```shell
N/A
```

### Output of `brew doctor`

```shell
N/A
```

### Description of issue

My sincerest apologies for asking such stupid questions. I am new to macOS and especially homebrew. I installed all apps using homebrew cask and I turned off automatic update inside app settings, e.g. Firefox - automatically install update. But still getting notification when updates arrive. I thought the experience will be similar to Linux like in case of update notification. My plan was to upgrade all apps on the weekend and continue working uninterruptedly the rest of the week. Could you please suggest what is the best practice for upgrading app (every day or weekly) and configure update settings for individual apps (auto update or turn off the update, ignore the notification till weekend)?

Update: I discovered the comprehensive explanation on https://docs.brew.sh/FAQ#why-arent-some-apps-included-during-brew-upgrade after SMillerDev's mention.

**Comment by SMillerDev** [comment, marked as answer]
https://joshtronic.com/2024/01/21/arch-linux-update-frequency/

This is about arch Linux, but the same principle applies. You can update as many times as you want.

---

You might want to check if Firefox still updates automatically though, since most Homebrew software that can auto update will be skipped by Homebrew updates.

**Reply by ghost** [reply]
Thank you for the reply and the amazing article. Could you please suggest any method to get notified of the vulnerabilities, like zero day for any app installed using Homebrew?
> since most Homebrew software that can auto update will be skipped by Homebrew updates.

Is it applicable for the auto-update capable apps or the auto-update is set to on or not?

**Reply by SMillerDev** [reply]
> Could you please suggest any method to get notified of the vulnerabilities, like zero day for any app installed using Homebrew?

There are none that I know of

> Is it applicable for the auto-update capable apps or the auto-update is set to on or not?

All software that Homebrew knows to be auto update capable will be skipped

**Reply by ghost** [reply]
> All software that Homebrew knows to be auto update capable will be skipped

OMG! I turned off all those auto-updates. Thank you so much for the information. Have a nice day.
